### PR TITLE
Remove unneeded API options from Form526 submit

### PR DESCRIPTION
**Pull request by sethdarragile6**
The PR removes some unneeded options from the call to Lighthouse submit

## Summary

- *This work is behind a feature toggle (flipper): YES*
- *(Summarize the changes that have been made to the platform)*
- *(If bug, how to reproduce)*
- *(What is the solution, why is this the solution?)*
- *(Which team do you work for, does your team own the maintenance of this component?)*
Disability Benefits Experience Team 1 - DBEX TREX 🦖
- *(If introducing a flipper, what is the success criteria being targeted?)*

## Related issue(s)

- *Link to ticket created in va.gov-team repo OR screenshot of Jira ticket if your team uses Jira*
- *Link to previous change of the code/bug (if applicable)*
- *Link to epic if not included in ticket*

## Testing done

- [ ] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: app/sidekiq/evss/disability_compensation_form/submit_form526.rb [hunk 1]
```diff
@@ -107,8 +107,7 @@ def perform(submission_id) # rubocop:disable Metrics/MethodLength
           begin
             submit_to_claims_api = submission.claims_api?
             # send submission data to either EVSS or Lighthouse (LH)
-            response = if Flipper.enabled?(:disability_compensation_lighthouse_submit_migration, user) ||
-                          submit_to_claims_api # right operand evaluation not needed once fully migrated to LH
+            response = if submit_to_claims_api # not needed once fully migrated to LH
```
Comment (freeheeling) [inline app/sidekiq/evss/disability_compensation_form/submit_form526.rb:110]: In addition to removing the `user` variable, there's no need to assign the `submit_to_claims_api` variable as the only other place it was used was as an argument in the now-defunct `generate_options_hash` method.

Comment (sethdarragile6) [reply app/sidekiq/evss/disability_compensation_form/submit_form526.rb]: ah thank you, I missed that! PR updated

File: app/sidekiq/evss/disability_compensation_form/submit_form526.rb [hunk 2]
```diff
@@ -103,7 +103,6 @@ def perform(submission_id) # rubocop:disable Metrics/MethodLength
           user_account = UserAccount.find_by(id: submission.user_account_id) ||
                          Account.find_by(idme_uuid: submission.user_uuid)
 
-          user = OpenStruct.new({ user_account_uuid: user_account.id, flipper_id: user_account.id })
           begin
             submit_to_claims_api = submission.claims_api?
```
Comment (tblackwe) [inline app/sidekiq/evss/disability_compensation_form/submit_form526.rb:107]: this can be combined with line 109, it is only used once.

Comment (tblackwe) [reply app/sidekiq/evss/disability_compensation_form/submit_form526.rb:107]: `submit_to_claims_api = submission.claims_api?`

Comment (sethdarragile6) [reply app/sidekiq/evss/disability_compensation_form/submit_form526.rb:107]: @tblackwe no, we determined that that assignment was important (@freeheeling identified this on [thread](https://dsva.slack.com/archives/C04U3EDUR3M/p1718224306419789?thread_ts=1718063256.233939&cid=C04U3EDUR3M). In fact I initially tried that and it was breaking tests.

Comment (freeheeling) [reply app/sidekiq/evss/disability_compensation_form/submit_form526.rb:107]: @sethdarragile6, that was before `options = generate_options_hash(submit_to_claims_api, user)` was removed in this PR. The assigned variable isn't needed as that was the only other instance in which it was used.

Comment (sethdarragile6) [reply app/sidekiq/evss/disability_compensation_form/submit_form526.rb:107]: ohhhhh I get now. Sorry, guys. Fixing now...

File: app/sidekiq/evss/disability_compensation_form/submit_form526.rb [hunk 3]
```diff
@@ -104,9 +104,8 @@ def perform(submission_id) # rubocop:disable Metrics/MethodLength
                          Account.find_by(idme_uuid: submission.user_uuid)
 
           begin
-            submit_to_claims_api = submission.claims_api?
             # send submission data to either EVSS or Lighthouse (LH)
-            response = if submit_to_claims_api # not needed once fully migrated to LH
+            response = if submit_to_claims_api? # not needed once fully migrated to LH
```
Comment (freeheeling) [inline app/sidekiq/evss/disability_compensation_form/submit_form526.rb:108]: The variable, `submit_to_claims_api`, isn't defined. You'll need `submission.claims_api?`

**Review by freeheeling — DISMISSED** [review]
🙌🏼  :shipit:

**Review by freeheeling — APPROVED** [review]

**Pull request merged**